packages/markers/src/common/problem-marker.ts

```diff
@@ -25,6 +26,6 @@ export interface ProblemMarker extends Marker<Diagnostic> {
 
 export namespace ProblemMarker {
     export function is(node: Marker<object>): node is ProblemMarker {
-        return 'kind' in node && node.kind === PROBLEM_KIND;
+        return isObject<ProblemMarker>(node) && node.kind === PROBLEM_KIND;
     }
```

Comment:
The fact that we have to use `isObject` here to determine that the `node` is an object implies that the `Marker<object>` contract doesn't hold up as expected. We should use `node: unknown` instead.